catalog/src/main/java/it/eng/catalog/service/DatasetService.java

```diff
@@ -153,4 +175,9 @@ public Artifact getArtifactFromDataset(String id) {
 		}
 		return artifact;
 	}
+
+	public Dataset getDatasetByArtifactForApi(String id) {
+        return repository.findByArtifact(id).orElse(null);
```

Comment:
Some special reason why returning here null and not Optional.empty as default behavior?